client/landing/gutenboarding/components/domain-picker-button/index.tsx

```diff
@@ -76,20 +76,28 @@ const DomainPickerButton: FunctionComponent< Props > = ( {
 	const [ isDomainPopoverVisible, setDomainPopoverVisibility ] = useState( false );
 	const [ isDomainModalVisible, setDomainModalVisibility ] = useState( false );
 
+	const handlePopoverOpen = () => {
+		recordEnterModal( 'DomainPickerPopover' );
+	};
+
 	const handlePopoverClose = ( e?: React.FocusEvent ) => {
 		// Don't collide with button toggling
 		if ( e?.relatedTarget === buttonRef.current ) {
 			return;
 		}
 		setDomainPopoverVisibility( false );
+		recordCloseModal( 'DomainPickerPopover', {
+			selected_domain: getSelectedDomain()?.domain_name,
+		} );
 	};
 
-	const handleModalClose = () => {
-		setDomainModalVisibility( false );
+	const handleModalOpen = () => {
+		recordEnterModal( 'DomainPickerModal' );
 	};
```

Comment:
Firing this event from the button doesn't make much sense. We'll open the modal also from the PlansGrid in the following days. I think it's ok to do this from modal/popover wrappers instead. That will also allow us to differentiate between type of modal opened.

Reply:
I agree! I'm going to move the firing of events out from DomainPickerButton and into the DomainPickerPopover and DomainPickerModal component individually.